client/src/app/shared/video/video.service.ts

```diff
@@ -115,6 +117,66 @@ export class VideoService {
       .catch((res) => this.restExtractor.handleError(res))
   }
 
+  baseFeed (
+    videoPagination: ComponentPagination,
+    sort: SortField
+  ): object {
+    const pagination = this.restService.componentPaginationToRestPagination(videoPagination)
+
+    let params = new HttpParams()
+    params = this.restService.addRestGetParams(params, pagination, sort)
+
+    let feed = {}
+    for (let item in FeedFormat) {
+      feed[FeedFormat[item]] = VideoService.BASE_FEEDS_URL + item.toLowerCase() + '?' + params.toString()
+    }
+
+    return feed
+  }
+
+  getFeed (
+    videoPagination: ComponentPagination,
+    sort: SortField,
+    filter?: VideoFilter
+  ): object {
+    let params = new HttpParams()
+    params = this.restService.addRestGetParams(params)
+
+    if (filter) {
+      params = params.set('filter', filter)
+    }
+
+    let feed = this.baseFeed(videoPagination, sort)
+    for (let item in feed) {
```

Comment:
`let item of feed`

Reply:
it doesn't work, as it yields an error that type '{}' is not of type array or chain.